src/helper/addProductFilters.ts

```diff
@@ -0,0 +1,68 @@
+import { Request } from "express";
+import { SelectQueryBuilder , sql} from "kysely";
+import { z } from "zod";
+import { DB } from "../database"; 
+
+export function addProductFilters(
+  req: Request,
+  query: SelectQueryBuilder<DB, "product", {}>
+) {
+  // Filter by product_id (number)
```

Comment:
z.coerce.number().safeParse(req.query.product_id).success
 
do this operation separately and use a variable with proper naming to understand its action.